src/output.ts

```diff
@@ -198,12 +198,16 @@ export default class Output {
 					const chars = styledCharsFromTokens(tokenize(line));
 					let offsetX = x;
 					for (const char of chars) {
+						if (offsetX >= this.width) break;
```

Comment:
How did you run into this issue where an out-of-bounds write would occur? Could you add a test for it to ensure it doesn't regress back?

Reply:
Done. In my case it was a pretty complicated nested layout, but I managed to reduce it to a `<Text wrap="end">` that's longer than the stdout width.

Reply:
There's no `wrap="end"` value, but there's `wrap="truncate-end"`. I updated it in the test you added, then commented out "prevent out-of-bounds writes" commit changes and it didn't crash for me. Looks like we don't need this code then?

Reply:
Hmm maybe the test case isn't correct then. I ended up with gaps in the output array, which crashed the function that converts Tokens back to strong.

Reply:
Do you still see the same issue when using `wrap="truncate-end"` instead of `wrap="end"` (which is an invalid prop)?

Reply:
Will try tomorrow or so.

Reply:
Hey @AlCalzone, do you still have the capacity to finish this PR? Would really love to ship it. No worries if you're busy though.

Reply:
No worries, take your time 👍